cmd/simRecon/simRecon.go

```diff
@@ -51,33 +51,59 @@ func ReconstructSeq(newickInput string, fastaInput string, outputFilename string
 	fasta.Write(outputFilename, treeFastas)
 }
 
-//SimRecon simulates evolution, performs reconstruction, and then evaluates the accuracy of the reconstruction
-func SimRecon(rootFastaFile string, treeFile string, gp string, simOutFile string, leafOutFile string, reconOutFile string, accuracyOutFile string) {
+//SimRecon simulates evolution, performs reconstruction, and then evaluates the accuracy of the reconstruction in two ways
+//default accuracy calculation will calculate both exonic and non-exonic accuracy for each node, and it's total accuracy
+//if there is a specified baseAccFile, this function will also return a file that contains 3 numbers:
+//the accuracy for all nodes for the first, second and third base of every codon
+func SimRecon(rootFastaFile string, treeFile string, gp string, simOutFile string, leafOutFile string, reconOutFile string, accuracyOutFile string, baseAccFile string) {
 	//TODO: this code will need to change drastically for sequences of varying lengths.
 	//The loop through the sequence is restricted by the length of a single fasta and the tot calculation will need to calculate the total number of bps
 	//ReconAccuracy calculates the total number of incorrectly reconstructed base pairs in a tree and returns a percentage of correct base calls
 	SimulateEvolve(rootFastaFile, treeFile, gp, simOutFile, leafOutFile)
 	ReconstructSeq(treeFile, leafOutFile, reconOutFile)
+	var calcBaseAcc = false
+	if baseAccFile != "" {
+		calcBaseAcc = true
+	}
 
-	answer := reconstruct.ReconAccuracy(simOutFile, reconOutFile, leafOutFile, gp)
+	answer, byBaseAnswer := reconstruct.ReconAccuracy(simOutFile, reconOutFile, leafOutFile, gp, calcBaseAcc)
 	out := fileio.EasyCreate(accuracyOutFile)
 	defer out.Close()
 
 	for name, accuracy := range answer {
 		fmt.Fprintf(out, "%s\t%f\n", name, accuracy)
 	}
+	if baseAccFile != "" {
+		baseAccOut := fileio.EasyCreate(baseAccFile)
+		defer baseAccOut.Close()
```

Comment:
Should probably do an explicit close at the end of the function and check for errors. EasyWriters have some pretty serious errors.